### PR TITLE
WakeUp for only a Single Command ("Amy Listen")

### PR DESCRIPTION
**Pull request by MakinZeel**
Add a new Wakeup Command for the Speech Recognition,
the SR shall just be awake for a Single Command

Issues
#283 

new Wake Up: "Hey Amy"

Pull request status:
- [ ] Tests added/updated
- [ ] Doku required and added
- [x] Ready for review

### CODE REVIEW

File: core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/natlang/JSGFGenerator.java [hunk 1]
```diff
@@ -51,14 +52,17 @@
...
 	 *            the wakeup command
+	 * @param singleWakeUp
+	 *            wakeup command for a single command
 	 * @param sleep
 	 *            the sleep command
 	 * @param shutdown
 	 *            the shutdown command
 	 */
-	public JSGFGenerator(String name, String wakeup, String sleep, String shutdown) {
+	public JSGFGenerator(String name, String wakeup, String singleWakeUp, String sleep, String shutdown) {
```
Comment (FlxB2) [inline core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/natlang/JSGFGenerator.java]: rename the wakeup variable, "wakeup" and "singleWakeUp" are not distinguished enough by name

File: core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/handler/LocalMainGrammarResultHandler.java [hunk 2]
```diff
@@ -50,19 +50,25 @@ public LocalMainGrammarResultHandler(SpeechRecognitionResultManager srManager, G
 	 */
 	@Override
 	protected boolean environmentSpecificInputHandling(String result, SpeechRecognitionResultManager srManager) {
-		if (result.equals(Constants.WAKE_UP)) {
-			srManager.handleListeningState(true);
-			srManager.voiceOutput("waking up");
+		switch (result) {
```
Comment (FlxB2) [inline core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/handler/LocalMainGrammarResultHandler.java]: the check srManage.isListening() can be checked before the switch occures

Comment (FlxB2) [reply core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/handler/LocalMainGrammarResultHandler.java]: in every case you check the same thing: srManager.isListening()

Comment (FlxB2) [reply core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/handler/LocalMainGrammarResultHandler.java]: in every case you check for the same thing: srManager.isListening()

Comment (FlxB2) [reply core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/handler/LocalMainGrammarResultHandler.java]: your right. This won't make much sense.

File: core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/output/OutputImpl.java [hunk 3]
```diff
@@ -122,14 +115,31 @@ public void run() {
...
-		} catch (SynthesisException e) {
-			this.logger.error("output error", e);
-		} catch (IOException e) {
-			this.logger.error("Error from MaryTTS InputStream", e);
+	private void writeAudio(OutputObject output) {
+		switch (output.getType()) {
+		case VOICE:
+			try {
+				moveBytes(this.tts.getMaryAudio(output.getMessage()), this.outputLine);
+			} catch (NullPointerException e) {
```
Comment (Legion2) [inline core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/output/OutputImpl.java]: don't catch NullPointerExceptions, do a null check

**Pull request merged**